### ISSUE TITLE
inter-project dependencies have no source mapping when using project views

### ISSUE THREAD

**Issue by lukaszwawrzyk** [JetBrains/bazel-bsp#121]
1. Import project view, for example `//server/src/main/java/org/jetbrains/bsp/bazel/server/bep/...:all`. 
2. Navigate to class from //commons like Constants. 
3. It takes to a class file inside a jar that is automatically decompiled, while project tree has actual source.

It should take to the original source file so that user can edit the file.

**Comment by abrams27** [comment]
i think that it is more a client side bug - the way how modules / dependencies are created in the intellij

**Comment by jastice** [comment]
Perhaps the source in project tree is not listed as dependency of the usage site. It might need to be referenced explicitly.

**Comment by lukaszwawrzyk** [comment]
@jastice Yes, this is indeed the case, I am looking for better user experience when dependency is not listed explicitly.

In fastpass inter-project dependencies were compiled to jars and additionally we prepared source jars that were linked to these sources. Thanks to that, user had just a slice of project imported and editable from IntelliJ. But still user was able to navigate through the code in the whole workspace. If his imported project depends on something else that is not directly imported, he could still jump to that class and read sources to understand how it works.
We added a special feature to pants plugin that shows a yellow bar on top of editor in such source dependency. It would ask user if file that owns this target should be also imported. In that case we were modifying list of imported targets and triggering sync again.
This whole solution together allowed users to navigate code without issues and easy way to allow editing such code if needed. The problem was that the extra sync could be quite long. I'd say this solution is acceptable and could maybe be ported to bazel, but maybe it is possible to do it better? Maybe we could attach sources directly from the repo to these jars?
